hedera-node/src/main/java/com/hedera/services/store/TopicStore.java

```diff
@@ -63,10 +67,80 @@ public void persistNew(Topic model) {
 	}
 
 	/**
-	 * Maps properties between a model {@link Topic} and a {@link MerkleTopic}
+	 * Persists the changes made to a {@link Topic} model into the state,
+	 * by mapping those changes onto a {@link MerkleTopic}.
+	 *
+	 * @param topic
+	 * 		- the model to be persisted
+	 */
+	public void persistTopic(Topic topic) {
+		final var id = topic.getId().asEntityNum();
+		final var mutableMerkleTopic = topics.get().getForModify(id);
+		mapModelToMerkle(topic, mutableMerkleTopic);
+		transactionRecordService.includeChangesToTopic(topic);
+	}
+
+	/**
+	 * Performs loading of a given merkle topic from state. The merkle topic is mapped to a model.
+	 * Please note that the model should be explicitly persisted with {@link TopicStore#persistTopic}
+	 * in order to apply the changes made.
+	 *
+	 * @param id
+	 * 		- model ID of the topic to be loaded.
+	 * @return {@link Topic} - the loaded topic
+	 */
+	public Topic loadTopic(Id id) {
+		final var entityNum = id.asEntityNum();
+		final var currentTopics = topics.get();
+		final var merkleTopic = currentTopics.get(entityNum);
+		final var topic = new Topic(id);
+
+		validateUsable(merkleTopic);
+		mapMerkleToModel(merkleTopic, topic);
+
+		return topic;
+	}
+
+	/**
+	 * Validates whether the given {@link MerkleTopic} is usable for operations.
 	 * 
-	 * @param model - the Topic model which will be used to map into a MerkleTopic
-	 * @param merkle - the merkle topic
+	 * @param topic - the topic to be validated
+	 */
+	private void validateUsable(MerkleTopic topic) {
+		validateFalse(topic == null, ResponseCodeEnum.INVALID_TOPIC_ID);
+		validateFalse(topic.isDeleted(), ResponseCodeEnum.INVALID_TOPIC_ID);
```

Comment:
If the topic is deleted, do we return `TOPIC_DELETED` or `INVALID_TOPIC_ID`?

Reply:
It's expected to be `INVALID_TOPIC_ID`, as in `TopicDeleteSuite#deleteWithAdminKey`

Reply:
Also:
```java 
@Override
	public ResponseCodeEnum queryableTopicStatus(TopicID id, MerkleMap<EntityNum, MerkleTopic> topics) {
		MerkleTopic merkleTopic = topics.get(EntityNum.fromTopicId(id));

		return Optional.ofNullable(merkleTopic)
				.map(t -> t.isDeleted() ? INVALID_TOPIC_ID : OK)
				.orElse(INVALID_TOPIC_ID);
	}
```